diff --git a/test/TestAdoption.sol b/test/TestAdoption.sol
index c8b6f62..38d09f5 100644
--- a/test/TestAdoption.sol
+++ b/test/TestAdoption.sol
@@ -7,4 +7,12 @@ import "../contracts/Adoption.sol";
 contract TestAdoption {
   Adoption adoption = Adoption(DeployedAddresses.Adoption());
 
+  // Testing the adopt() function
+  function testUserCanAdoptPet() public {
+    uint returnedId = adoption.adopt(8);
+
+    uint expected = 8;
+
+    Assert.equal(returnedId, expected, "Adoption of pet ID 8 should be recorded.");
+  }
 }
