diff --git a/test/TestAdoption.sol b/test/TestAdoption.sol
index 38d09f5..a5c1f21 100644
--- a/test/TestAdoption.sol
+++ b/test/TestAdoption.sol
@@ -15,4 +15,14 @@ contract TestAdoption {
 
     Assert.equal(returnedId, expected, "Adoption of pet ID 8 should be recorded.");
   }
+
+  // Testing retrieval of a single pet's owner
+  function testGetAdopterAddressByPetId() public {
+    // Expected owner is this contract
+    address expected = this;
+
+    address adopter = adoption.adopters(8);
+
+    Assert.equal(adopter, expected, "Owner of pet ID 8 should be recorded.");
+  }
 }
