diff --git a/test/TestAdoption.sol b/test/TestAdoption.sol
index 346db32..e666dda 100644
--- a/test/TestAdoption.sol
+++ b/test/TestAdoption.sol
@@ -27,4 +27,12 @@ contract TestAdoption {
 
     Assert.equal(adopter, expectedAdopter, "Owner of the expected pet should be this contract");
   }
+
+  // Testing retrieval of all pet owners
+  function testGetAdopterAddressByPetIdInArray() public {
+    // Store adopters in memory rather than contract's storage
+    address[16] memory adopters = adoption.getAdopters();
+
+    Assert.equal(adopters[expectedPetId], expectedAdopter, "Owner of the expected pet should be this contract");
+  }
 }
