diff --git a/src/js/app.js b/src/js/app.js
index daf0b4b..4f75ffe 100644
--- a/src/js/app.js
+++ b/src/js/app.js
@@ -49,10 +49,17 @@ App = {
   },
 
   initContract: function() {
-    /*
-     * Replace me...
-     */
+    $.getJSON('Adoption.json', function(data) {
+      // Get the necessary contract artifact file and instantiate it with truffle-contract
+      var AdoptionArtifact = data;
+      App.contracts.Adoption = TruffleContract(AdoptionArtifact);
 
+      // Set the provider for our contract
+      App.contracts.Adoption.setProvider(App.web3Provider);
+
+      // Use our contract to retrieve and mark the adopted pets
+      return App.markAdopted();
+    });
     return App.bindEvents();
   },
 
